src/main/kotlin/interactor/GetSalaryAverageAndCitiesNamesInCountryInteractor.kt

```diff
@@ -1,28 +1,23 @@
 package interactor
 
-import model.CityEntity
+import utils.isNotNull
 
 
 class GetSalaryAverageAndCitiesNamesInCountryInteractor(
     private val dataSource: CostOfLivingDataSource
     ) {
 
     fun execute(countryName:String):List<Pair<String,Float>> {
-        return if (checkIfCountryIsValid(countryName))
-            dataSource.getAllCitiesData()
-                .filter{excludeNullSalariesAndLowQualityData(it) && it.country.lowercase() == countryName.lowercase()}
-                .map{Pair(it.cityName,it.averageMonthlyNetSalaryAfterTax!!)}
-        else
-            throw Exception("Invalid country name")
-    }
-
-    private fun checkIfCountryIsValid(countryName: String):Boolean{
-        return countryName.lowercase() in dataSource.getAllCitiesData()
-            .map{it.country.lowercase()}
-    }
+        return dataSource
+            .getAllCitiesData()
```

Comment:
To make sure about the returned data from the data source, check if it is an empty list or not, if it is then throw IllegalStateException("Something went wrong")
```suggestion
            .getAllCitiesData()
            .let{ it.ifEmpty{ IllegalStateException("Something went wrong") } }
```

Reply:
you say, that we should not return exception if the list is empty. why here you suggested the oppisite?